flare/output.py

```diff
@@ -2,204 +2,320 @@
 import datetime
 import numpy as np
 import multiprocessing
+import os
+import shutil
 
 
-def write_to_output(string: str, output_file: str = 'otf_run.out'):
-    with open(output_file, 'a') as f:
-        f.write(string)
-
+class Output():
+    """
+    Output class host the logfile, xyz config file and force xyz file.
+    :param basename
+    """
 
-def write_header(cutoffs, kernel_name, hyps, algo, dt, Nsteps, structure,
-                 output_name, std_tolerance,
-                 optional: dict = None):
-    with open(output_name, 'w') as f:
+    def __init__(self, basename: str = 'otf_run'):
+        """
+        open files with the basename and different suffices.
+        """
+        self.basename = "{}".format(basename)
+        self.outfiles = {}
+        filesuffix = {'log': '.out', 'xyz': '.xyz', 'fxyz': '-f.xyz', 'hyps':"-hyps.dat"}
+
+        for filetype in filesuffix.keys():
+            self.open_new_log(filetype, filesuffix[filetype])
+
+    def conclude_run(self):
+        """
+        destruction function that close all files
+        """
+        print('-'*20, file=self.outfiles['log'])
+        print('Run complete.', file=self.outfiles['log'])
+        for (k, v) in self.outfiles.items():
+            v.close()
+        del self.outfiles
+        self.outfiles = {}
+
+    def open_new_log(self, filetype, suffix):
+
+        filename = self.basename+suffix
+
+        # if the file exists, back up
+        if os.path.isfile(filename):
+            shutil.copy(filename, filename+"-bak")
+
+        if filetype in self.outfiles.keys():
+            if self.outfiles[filetype].closed:
+                self.outfiles[filetype] = open(filename, "w+")
+        else:
+            self.outfiles[filetype] = open(filename, "w+")
+
+    def write_to_log(self, logstring: str, name: str = "log"):
+        """
+        Write any string to logfile
+        """
+        self.outfiles[name].write(logstring)
+
+    def write_header(self, cutoffs, kernel_name,
```

Comment:
In the future, we should consider writing a more general function that takes a dictionary in as an argument and then prints the key, value pairs in a similar way that the header method does (as well as a few other methods in this document which are tailored to individual use cases).